frontend/src/components/creditas-form-container/creditas-form-container.component.js

```diff
@@ -0,0 +1,193 @@
+    const TYPE = loanTypesConfig.types[this.loanType]
+
+    this.$warrantyRange.setAttribute('initial-value', TYPE.minWarrantyAmount)
+    this.$warrantyRange.setAttribute('min-value', TYPE.minWarrantyAmount)
+    this.$warrantyRange.setAttribute('max-value', TYPE.maxWarrantyAmount)
+    this.$loanRange.setAttribute('initial-value', TYPE.minLoanAmount)
+    this.$loanRange.setAttribute('min-value', TYPE.minLoanAmount)
+    this.$loanRange.setAttribute('max-value', TYPE.minLoanAmount)
+  }
+
+  /**
+   * Sets all event listeners.
+   *
+   * @memberof CreditasFormContainer
+   */
+  setEventListeners () {
+    this.addEventListener('creditas-dropdown:changed', this.onDropdownChange, false)
+    this.addEventListener('creditas-range-input:changed', this.onRangeChange, false)
+    this.addEventListener('creditas-cta:clicked', this.onCtaClicked, false)
+  }
+
+  /**
+   * Removes all event listeners.
+   *
+   * @memberof CreditasFormContainer
+   */
+  removeEventListeners () {
+    this.removeEventListener('creditas-dropdown:changed', this.onDropdownChange, false)
+    this.removeEventListener('creditas-range-input:changed', this.onRangeChange, false)
+    this.removeEventListener('creditas-cta:clicked', this.onCtaClicked, false)
+  }
+
+  /**
+   * Called when the warranty dropdown changes its value.
+   *
+   * @param {*} event
+   * @memberof CreditasFormContainer
+   */
+  onDropdownChange (e) {
+    if (e.detail && e.detail.label === 'garantia') {
+      // Updating the months dropdown if the type is changed.
+      const MONTHS = loanTypesConfig.types[e.detail.value].months.join(',')
+      this.$selectMonths.setAttribute('dropdown-values', MONTHS)
+
+      // Updating the selected loan type and the ranges values accordingly.
+      this.loanType = e.detail.value
+      this.loanMonths = loanTypesConfig.types[this.loanType].months[0]
+      this.updateInitialRangeValues()
+    }
+
+    if (e.detail && e.detail.label === 'parcelas') {
+      this.loanMonths = e.detail.value
+    }
+
+    this.updateSimulationBox()
+  }
+
+  /**
+   * Triggered when the form submit is triggered.
+   *
+   * @param {*} event
+   * @memberof CreditasFormContainer
+   */
+  onCtaClicked (event) {
+    let values = getFormValues(this.$form)
+    let confirmation = `Confirmação\n${values
+      .map(value => `Campo: ${value.field}, Valor: ${value.value}`)
+      .join('\n')}`.concat(
+        `\nTotal ${toCurrency((IOF + INTEREST_RATE + this.loanMonths + 1) * this.loanAmount)}`
+      )
+
+    alert(confirmation)
+  }
+
+  /**
+   * Updates the max loan range, depending on the warranty amount selected.
+   * The loan a customer can get access to is never higher than 80% of the warranty provided.
+   *
+   * @param {*} e
+   * @memberof CreditasFormContainer
+   */
+  onRangeChange (e) {
+    if (e.detail && e.detail.label === 'garantia') {
+      this.$loanRange.setAttribute('max-value', e.detail.value * loanTypesConfig.maxLoanRate)
```

Comment:
This will update the loan amount range when the warranty range changes, to only allow a maximum amount of 80% of the provided warranty. That way we wouldn't need validation at all and the user does not have to know about complicated rules :)